### PR TITLE
[docs] Add RTL documentation for the pickers

### LINKED ISSUE #11523: [docs] Add Pickers RTL support section
Created from: https://github.com/mui/mui-x/pull/6636#issuecomment-1870134920

> Do we have a docs section about RTL?

>I found [this](https://mui.com/x/react-date-pickers/base-concepts/#field-components) about extra unicode characters, but no section about how to setup RTL like for the data grid https://mui.com/x/react-data-grid/localization/#rtl-support

**Search keywords**:

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #11523

- [x] Unify how we set the RTL in all the demos of hijri and jalali adapters (I took the data grid RTL demo as source of truth, it manually sets the `rtlPlugin` from stylis)
- [x] Add a new doc section + demo on the localization doc to introduce the RTL support
- [x] Fix the `dir="rtl"` on pickers (the view are inside a portal, so they were not impacted by the `dir="rtl"` set in the demo)

### CODE REVIEW

File: docs/data/date-pickers/calendar-systems/AdapterHijri.tsx [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
+  stylisPlugins: [prefixer, rtlPlugin],
+});
 
 export default function AdapterHijri() {
+  // Inherit the theme from the docs site (dark/light mode)
   const existingTheme = useTheme();
+
   const theme = React.useMemo(
-    () => createTheme({ direction: 'rtl' }, existingTheme),
+    () => createTheme(existingTheme, { direction: 'rtl' }),
```
Comment (flaviendelangle) [inline docs/data/date-pickers/calendar-systems/AdapterHijri.tsx:23]: The order was wrong, `existingTheme` contains `direction"=ltr"` so it overrode `{ direction: 'rtl' }`

This, plus the missing stylis plugin meant that our UI was basically a mix of LTR and RTL on those demos.

File: packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx [hunk 2]
```diff
@@ -397,11 +397,11 @@ export const MultiSectionDigitalClock = React.forwardRef(function MultiSectionDi
       return views;
     }
     const digitViews = views.filter((v) => v !== 'meridiem');
-    const result: TimeViewWithMeridiem[] = digitViews.toReversed();
+    digitViews.reverse();
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx:400]: `.toReversed` is too recent, I avoid using it.
It was causing the following issue in the CI when building the doc:

```
9:25:02 AM: TypeError: ec.filter(...).toReversed is not a function
```

Comment (JCQuintas) [reply packages/x-date-pickers/src/MultiSectionDigitalClock/MultiSectionDigitalClock.tsx:400]: Circle CI is on Node 18, `toReversed` is Node 20.

File: docs/data/date-pickers/adapters-locale/adapters-locale.md [hunk 3]
```diff
@@ -369,3 +369,12 @@ moment.updateLocale('en', {
   },
 });
 ```
+
+## RTL Support
+
+Right-to-left languages such as Arabic, Persian, or Hebrew are supported.
+Follow [this guide](/material-ui/customization/right-to-left/) to use them.
```
Comment (arthurbalduini) [inline docs/data/date-pickers/adapters-locale/adapters-locale.md:376]:
```suggestion
Follow [this guide](https://mui.com/material-ui/customization/right-to-left/) to use them.
```

Comment (flaviendelangle) [reply docs/data/date-pickers/adapters-locale/adapters-locale.md:376]: Why should we make this URL absolute?
It will be deployed in the same domain

Comment (arthurbalduini) [reply docs/data/date-pickers/adapters-locale/adapters-locale.md:376]: My bad, I didn't notice this was on material-ui docs 🙈

**Review by arthurbalduini — APPROVED** [review]
🎉

**Pull request merged**